### ISSUE TITLE
wc_stripe_payment_request_total_label_suffix not working

### ISSUE THREAD

**Issue by giannisftaras** [woocommerce/woocommerce-gateway-stripe#1434]
**Describe the bug**<br>Stripe - Woocommerce Docs lists the wc_stripe_payment_request_total_label_suffix function as a way to change the *via Woocommerce* label on Apple/Google Pay. <br>Adding `add_filter( 'wc_stripe_payment_request_total_label_suffix', '__return_null');` to the functions.php file doesn't do anything.<br>Stripe - Woocommerce Docs lists the wc_stripe_payment_request_total_label_suffix function as a way to change the *via Woocommerce* label on Apple/Google Pay. <br>Adding `add_filter( 'wc_stripe_payment_request_total_label_suffix', '__return_null');` to the functions.php file doesn't do anything.

**To Reproduce**<br>Steps to reproduce the behavior:<br>Steps to reproduce the behavior:

1. Add *wc_stripe_payment_request_total_label_suffix* filter to functions.php
2. Make corresponding function return null or any other string
3. Go to cart page and click on the Google/Apple Pay button. The *via Woocommerce* label will still be there.

**Expected behavior**<br>The  *wc_stripe_payment_request_total_label_suffix* should return a string which will be added at the end of the Statement Descriptor.<br>The  *wc_stripe_payment_request_total_label_suffix* should return a string which will be added at the end of the Statement Descriptor.

**Environment (please complete the following information):**

* WordPress Version 5.6
* WooCommerce Version 4.8.0
* Stripe Plugin Version 4.7.0
* Browser \[e.g. chrome, safari\] and Version Chrome 87.0.4280.88

**Comment by iamkrunalprajapati** [comment]
@giannisftaras Yes, I am also facing the same issue for this, it looks like there is no effect if we try to override that label via hook

**Comment by davidrgz2021** [comment]
Hi!
Anyone could sove this problem?  How to hide "via woocommerce" from Google Pay?

**Comment by qas1** [comment]
For anyone still stuck on this, this seems to only work when you set the add_filter before the plugin is loaded. I've manged this by adding the code to the mu-plugins directory and it works fine.

https://wordpress.org/support/article/must-use-plugins/

**Comment by iamkrunalprajapati** [comment]
@qas1 Yes with mu-plugins it has been resolved, Thank you!

**Comment by wjrosa** [comment]
Closing as resolved. The Payment Request Button implementation this was reported against has been removed and replaced by the Express Checkout Element. In the current code the `wc_stripe_payment_request_total_label_suffix` filter is applied in `WC_Stripe_Express_Checkout_Helper`, so the suffix can be changed or removed. If the filter misbehaves on a current version, please open a fresh issue.

> *Drafted with AI; reviewed by me.*
